quesma/queryparser/aggregation_date_range_parser.go

```diff
@@ -3,80 +3,53 @@
 package queryparser
 
 import (
-	"quesma/logger"
+	"fmt"
 	"quesma/model/bucket_aggregations"
 	"unicode"
 )
 
-func (cw *ClickhouseQueryTranslator) parseDateRangeAggregation(dateRange QueryMap) (bucket_aggregations.DateRange, error) {
-	var err error
-	var fieldName, format string
-
-	if field, exists := dateRange["field"]; exists {
-		if fieldNameRaw, ok := field.(string); ok {
-			fieldName = cw.ResolveField(cw.Ctx, fieldNameRaw)
-		} else {
-			logger.WarnWithCtx(cw.Ctx).Msgf("field specified for date range aggregation is not a string. Using empty. Querymap: %v", dateRange)
-		}
-	} else {
-		logger.WarnWithCtx(cw.Ctx).Msgf("no field specified for date range aggregation. Using empty. Querymap: %v", dateRange)
-	}
-	var ranges []any
-	var ok bool
-	if formatRaw, exists := dateRange["format"]; exists {
-		if formatParsed, ok := formatRaw.(string); ok {
-			format = formatParsed
-		} else {
-			logger.WarnWithCtx(cw.Ctx).Msgf("format specified for date range aggregation is not a string. Using empty. Querymap: %v", dateRange)
-		}
+func (cw *ClickhouseQueryTranslator) parseDateRangeAggregation(aggregation *pancakeAggregationTreeNode, params QueryMap) (err error) {
```

Comment:
I wonder about type signature? Why pass `aggregation` as argument, shouldn't we just return optional queryType?

Reply:
Ok, after reading later it become obvious why you did that way.